client/src/app/employee/employee.component.ts

```diff
@@ -10,11 +16,30 @@ import { faLock, faUnlock } from '@fortawesome/free-solid-svg-icons';
 export class EmployeeComponent {
   readonly faLock = faLock;
   readonly faUnlock = faUnlock;
-  isReadonly = true;
+  employees: Observable<EmployeeList[]>;
+  employee: Observable<Employee>;
+  uiState = '';
+
+  constructor(
+    private employeeDetailQuery: EmployeeDetailQuery,
+    private employeeDetailService: EmployeeDetailService,
+    private employeeListService: EmployeeListService,
+    private employeeListQuery: EmployeeListQuery,
+  ) {
+    this.employee = this.employeeDetailQuery.select(
+      (state) => state as Employee,
+    );
 
-  constructor() {}
+    this.employees = this.employeeListQuery.selectAll();
+    this.employeeListService.getEmployees();
+  }
+
+  setCreate() {
+    this.uiState = 'create';
+  }
 
-  toggleReadonly() {
-    this.isReadonly = !this.isReadonly;
+  setActive(employee: EmployeeList) {
+    this.employeeDetailService.getEmployee(employee.id);
```

Comment:
These two operations would go well together. These are two state changes (setting the `uiState` and populating the `Employee`) that necessarily both have to happen for the application to work correctly.

This is an example where moving local state into an Akita store and having the `EmployeeDetailService` both request the new `Employee` data and change the `uiState` in the store, would be a good approach.

It's fine to leave it as it is for now though!

Reply:
Definitely want to come back to this at some point, i haven't had an opportunity to store the UI specific state yet as all of uses for akita that i've came across have been strictly for sending/retrieving data to display in the app.